processor/baggage/lib/opentelemetry/processor/baggage/baggage_span_processor.rb

```diff
@@ -42,6 +47,16 @@ module Baggage
       #     )
       #   end
       class BaggageSpanProcessor < OpenTelemetry::SDK::Trace::SpanProcessor
+        # Create a new BaggageSpanProcessor that reads Baggage keys and values from the parent context
+        # and adds them as attributes to the span.
+        #
+        # @param [lambda] baggage_key_predicate A lambda that takes a baggage key and returns true if
+        #  the key should be added to the span as an attribute, false otherwise.
+        def initialize(baggage_key_predicate)
+          @baggage_key_predicate = baggage_key_predicate
```

Comment:
NIT: could `ALLOW_ALL_BAGGAGE_KEYS` be default for `@baggage_key_predicate`?

e.g.
```
def initialize(baggage_key_predicate: nil)
    @baggage_key_predicate = baggage_key_predicate || ALLOW_ALL_BAGGAGE_KEYS
    ...
end
```

Reply:
I went with a required user parameter to make it super clear to users they need to explicitly allow baggage keys they want. If we used the `ALLOW_ALL_BAGGAGE_KEYS ` predicate as the default value, it would be easy for users to unknowing include copy all baggage entires by default.

Using a required parameter helps make it more secure by default.

Reply:
I agree with the opt-in behavior of needing to specify a predicate method to the constructor.